src/models/TeamGenerator.ts

```diff
@@ -178,6 +178,20 @@ export default class TeamGenerator {
         return [];
     }
 
+    /**
+     * This function is a protected subroutine call which takes 2 full teams and 1 game as input
+     * and tries to swap 1 player pair if applicable to decrease the skill diff of the given teams
+     * as much as possible. If several options with the same result are available then a random
+     * pair out of the options is chosen.
+     * 
+     * NOTE!: this function does no param check as it is only a subroutine and expects the given teams
+     * to be full and the game skill is given for all players for the passed game.
+     * 
+     * @param team1 The first team to be balanced.
```

Comment:
Not sure if you want to add the type in the JSDoc as well? e.g. @param {Team} team1

Reply:
Ah nice, I did not know that (or not anymore). Thanks for the hint. Will do.

Reply:
I tried to add this but somehow in VS code this doesn't show in the documentation.